v2/contribute/decisions/reactsdk/0002-split-recipe-route-handlers-into-separate-comps.mdx

```diff
@@ -94,27 +96,27 @@ With react-router-dom:
 function App() {
   return (
     <SuperTokensWrapper>
-      <div className="App">
-        <Router>
-          <Routes>
-            <EmailPasswordPreBuiltUIRoutes router={require("react-router-dom")} />
-            <EmailVerificationPreBuiltUIRoutes
-              router={require("react-router-dom")}
-              componentOverrides={{
-                EmailVerificationSendVerifyEmail_Override: () => <OtpScreen />,
-              }}
-            />
-            <Route
-              path="/"
-              element={
-                <SessionAuth>
-                  <Home />
-                </SessionAuth>
-              }
-            />
-          </Routes>
-        </Router>
-      </div>
+      <EmailVerificationComponentsOverrideProvider
+        components={{
+          EmailVerificationSendVerifyEmail_Override: () => <OtpScreen />,
+        }}>
+        <div className="App">
+          <Router>
+            <Routes>
+              <EmailPasswordPreBuiltUIRoutes router={require("react-router-dom")} />
+              <EmailVerificationPreBuiltUIRoutes router={require("react-router-dom")} />
```

Comment:
can this be used to render these on a dynamic route? For example for different languages

Reply:
@porcellus missed this? Can you add more details about this - if we can / should do this via these components or some other method. Another way of doing this would be to make the websiteBasePath a regex  (which can be done later on as well)

Reply:
No, I just haven't had time to address this.
This method could be used to do this, but it'll take a bit further work. Right now, this would only match paths starting with websitebasepath.

Reply:
So let's keep it the way it is. We can address the issue of dynamic websiteBasepath later on